solr-updater/src/main/java/dk/dbc/search/solrdocstore/updater/Worker.java

```diff
@@ -128,4 +165,27 @@ public void destroy() {
         };
     }
 
+    public Runnable processForOneCollection(JsonNode sourceDoc, SolrCollection collection, String pid, Integer commitWithin) {
+        return () -> {
+            try (DBCTrackedLogContext logContext = new DBCTrackedLogContext()
+                    .with("pid", pid)
+                    .with("solrCollection", collection.getName())) {
+                log.info("Building document");
+                SolrInputDocument solrDocument = docProducer.createSolrDocument(sourceDoc, collection);
+                String bibliographicShardId = DocProducer.bibliographicShardId(sourceDoc);
+                int nestedDocumentCount = docProducer.getNestedDocumentCount(bibliographicShardId, collection);
+
+                docProducer.deleteSolrDocuments(bibliographicShardId, nestedDocumentCount, collection, commitWithin);
+
+                docProducer.deploy(solrDocument, collection, commitWithin);
+                int count = 1;
+                if (solrDocument != null && solrDocument.hasChildDocuments())
+                    count += solrDocument.getChildDocumentCount();
+                log.info("Deleted {} record(s) and added {} to SolR", nestedDocumentCount + 1, count);
+                docStasher.store(pid, solrDocument);
+            } catch (PostponedNonFatalQueueError | IOException | SolrServerException ex) {
+                throw new RethrowableException(ex);
```

Comment:
Jeg kender/forstår ikke fejlhåndteringen - men jeg har sådan en fornemmelse af at her ville være et OK sted at logge noget information om f.eks. pid, solrcollection eller andet information der kan spores til det dokument der fejlede.

Reply:
Den information er embedded i en cause exception... Ting brugt i Function<T, R> et. al. må kun kaste RuntimeException, Derfor Rethrowable, som har muligheden for at rethrow'e som den type der nu en gang er givet med som cause

Reply:
Ah, det giver mening.